extensions/workloads/template.go

```diff
@@ -42,7 +42,7 @@ func NewPodTemplate(containers []corev1.Container, volumes []corev1.Volume, imag
 }
 
 // NewDeploymentTemplate is a constructor that creates a deployment template. If the isCattleLabeled true, workloadselector labels are assigned to the deployment and the pod template.
-func NewDeploymentTemplate(deploymentName string, namespace string, template corev1.PodTemplateSpec, isCattleLabeled bool, matchLabels map[string]string) *appv1.Deployment {
+func NewDeploymentTemplate(deploymentName string, namespace string, template corev1.PodTemplateSpec, isCattleLabeled bool, matchLabels map[string]string, isWindows bool) *appv1.Deployment {
```

Comment:
since `template` is an input variable, wouldn't it make sense to add the Spec.NodeSelector before calling this NewDeploymentTemplate function? i.e. do the if isWindows statement in the NewPodTemplate function instead of here.

Reply:
I personally think it makes more sense to leave it as it is. When I did my comprehensive testing on my final commit and during release testing, multiple permutations for clusters had to be done. And having it at this level caused no issues in any of the cluster configurations.

Reply:
I 100% agree with Caleb, doesn't it make way more since to do this in NewPodTemplate instead? template is an input variable. Also make sure the constructor is appropriately updated in tests/actions that use this function

Reply:
Done.